### DISCUSSION TITLE
"Welcome to LXQt" Configurator Ideas

### DISCUSSION THREAD

**Discussion by AzumaHazuki** [lxqt/lxqt#2323] [Ideas]
I've been thinking a lot about that spooky "choose your WM" dialog that pops up on first LXQt start if it's installed from scratch like on Arch or Gentoo, or really any distro that hasn't pre-configured its environment. An idea came to me: why not have a sort of landing screen for first setup, like a much cleaner and less obtrusive version of what KDE 3.x used to have?

The vision for this includes:
- Page 1: the "Welcome to LXQt" page, with a short summary of what the project is, what the current screen is and why it's there, and the usual back/forward buttons in the lower-right corner. Sample Prompt:

"Welcome to LXQt! This is the first-time configuration dialogue that will help you customize your workflow in the way that best suits you. On the next few pages, you will be able to configure the layout, themes, icons, and window manager."

- Page 2: the Panel Layout page, which sets up the panels and widgets in them. I am envisioning 3 or 4 defaults here, with more of course possible; each would be a small thumbnail and radio button in the left of the window and a header and brief description to the right, as follows:
- - `[x] Traditional (or "Panel at Bottom"/"Bottom Panel" etc): "The familiar, default layout."`
- - `[ ] Two Panels (or "Classic"/"GTK 2"/"Linux"/etc): "Top and bottom panels, as in the golden age of Gnome and Xfce."`
- - `[ ] Top (or "Cupertino"/"OS X"/"MacOS"/"Top Panel" etc): "Panel at the top of the screen. Useful with an external dock application such as Plank or Latte-Dock."`
- - `[ ] Widescreen (or "Panel at Side"/"Unity"/etc): "A side-anchored panel with icons-only task bar. Makes good use of space on 16:9 displays."`

Under this, we should allow the user to choose an LXQt theme, and possibly set panel sizes from a few defaults. I'm thinking something like  
`[x] Normal - 32 pixels high, 24-pixel icons [ ] Slim - 24 pixels high, 16-pixel icons
`
- - Of course each layout is going to need a file in /usr/share/lxqt/lxqt-panel-$OPTION.conf that will be copied to the user's home directory. I can provide these by setting them up manually on my machine and uploading them somewhere.

- Page 3: the Application Themes/Fonts/Icons page. This may need to be split into 2 pages if it's too cluttered, but the gist of it is this is the page where users can set their font, icon theme, Qt theme, Qt palette, and GTK 2/3 themes. Sectioned as follows:
- - Font: choose font, size, antialiasing style, hinting level, subpixel rendering, and DPI. 
- - - I think good defaults are {Liberation Sans, Noto Sans, DejaVuSans, etc}, size 10, AA on, Slight hinting, RGB rendering, 96 DPI
- - Icon Theme: choose icon theme. LXQt already pulls in the Oxygen icons on Gentoo and I know Lubuntu has a Papirus-ish set. 
- - - Do we depend on any given icon theme? If so, Oxygen is a decent default.
- - Qt theme. Can default to Fusion as normal, and maybe we can bundle <a href=https://www.pling.com/p/1931943>Plastik</a> as well? I also think we should provide a palette for any LXQt themes we ship, and perhaps others; I may be able to, for example, extract the relevant 8 colors from the entirety of Trinity or Plasma's color schemes...maybe we could break this out into a package of its own, as lxqt-palettes or lxqt-colors. 
- - GTK 2 and 3 themes. This may need a little explanatory blurb as to what these are ("Some popular applications not part of LXQt, such as Firefox, use the GTK widget set. Use this section to select a theme for them. For a consistent look, you may choose the same theme for GTK2 and 3"). Should likely default to Adwaita.

- Page 4: Window manager selection. This page should do a check in the background for various WMs in $PATH and present a list, but needs some explanatory text up top. I'm thinking something like this:

> 
> **Window Manager Selection**
> 
> The _window manager_ is the program that draws titlebars and buttons on individual windows. LXQt defaults to using the Openbox window manager, but you may change this by checking the "custom window manager" box below and entering the full path to your preferred one. If you are not sure what to do here, the default is perfectly fine."

`[ ] Custom Window Manager? [ /usr/bin/openbox ] (text entry field greyed out unless box is checked)
`
> **Compositor Selection** 
>
> A _compositor_ is a program that allows for transparencies, fading, and other visual effects. LXQt defaults to Picom, a lightweight compositor with simple defaults. 
> 
> **Please note**:  some window managers have their own compositors. If you have chosen one of these above, it is important to leave the **No** box selected!"
> 
> 

`Use Compositor? [x] Yes [ ] No [ ] Custom: [ /usr/bin/picom ] (as above, greyed out unless box for custom option is checked)
`
- - If we use this as an option, should we just let Picom use its defaults, or ship a minimalistic picom.conf and silently point the compositor at it on startup? Some people might not like the slow fades and partial titlebar transparencies Picom defaults to...also, if Picom isn't installed we should warn the user about it.

- Page 5: Unless there are other things you think we should let people set (clock? timezone? language? keymap? and if so, maybe these should be on Page 2 and the theming stuff later?) this can be the last page. I think it should have the usual "All done! We hope you enjoy using LXQt" message, plus a reminder that the configurator can be run again later if needed.

Unfortunately, I can't code. The most I could do here is make mockups in GIMP or so, so if this sounds like something that would be useful to LXQt it's going to need other peoples' help @_@

**Comment by stefonarch** [comment]
I saw 2 similar things, one first run on Manjaro, the otherone on a distro called Emmabuntus. It's a nice thing to have, but distros should do it IMHO.

**Comment by tsujan** [comment]
WM is an important component; hence the dialog for choosing it.

Other components have their own settings. I don't see any reason to show a first-run dialog about panel, Qt style, icon theme, time,....Users are capable of finding the pertinent settings dialogs. The first-run dialog that LXQt shows should remain minimal; distros are free to add their own first-run dialogs.

**Comment by AzumaHazuki** [comment]
Then, at the very least, let's make that dialog friendlier! I have been using Linux for 18 years as of this July, almost all of it on Gentoo, and am only 34, and that screen _still_ spooks me out. It's going to scare off any new user, and while we hear much "this is the distro's job," most distros do no such thing.

LXQt has so, so much potential. GTK (due to GNOME) has gone completely insane and I truly believe Qt is going to be the way forward. But LXQt has such a serious "attitude problem" up front. 

I can't imagine such configurator would be very heavy; perhaps, we could settle for just the WM, compositor, and panel layout choices and allow the user to choose themes and so on later, though if we do that, we should implement some better defaults...

**Reply by tsujan** [reply]
> that screen still spooks me out.

On the contrary, a detailed first-run dialog would spook out many — to say nothing of its being redundant.

> LXQt has such a serious "attitude problem"

Among other things, LXQt tries to avoid duplicated functionalities. That "attitude" may be undesirable to some, as is the case with other DEs. Not all people love a specific DE.

**Reply by Butterfly** [reply]
> Then, at the very least, let's make that dialog friendlier! I have been using Linux for 18 years as of this July, almost all of it on Gentoo, and am only 34, and that screen _still_ spooks me out. It's going to scare off any new user, and while we hear much "this is the distro's job," most distros do no such thing.

I'm a very long time user of Linux, too, but I have PTSD triggered cognitive issues that actually help me function as a perennial newbie. An observation: Some packages have taken to adding tool tips to the various steps for using their software. That is SO user-friendly!

Just something simple, some are even a couple sentences, that explain what each step in using packages does. The always visual part of the screen can be a simple explanatory phrase that is then backed up by a more extended mouse-over tool tip. The caveat is that not all users, especially new ones, know that tool tips can occasionally be accessed via the cursor. :smile:

**Comment by darkshram** [comment]
@tsujan: My opinion is to keep the first-run dialog and even extend its capabilities. 

Also, I still thing is a good idea to try to avoid depending on third-party components. Will try to explain why:

Gtk4 just showed everyone GNOME do not care about third-party software. To MATE desktop it is **almost** a dead end, since most of the widgets used for the current interface are X11 specific and are marked as deprecated in current Gtk3 and are non-existent o soon-to-be-removed in Gtk4. Short long story: MATE can not migrate to Gtk4, unless they **re-write all** the interface using non-deprecated widgets. Once Gtk3 and X11 reach EOL, it will be the last nail of the coffin for MATE, unless they fork Gtk3 (which is something they will not do because they do not have the human resources to do it). Unfortunately for MATE desktop, it depends on Glib and Gtk, two core libraries controlled by GNOME and Red Hat (RH pays the salaries of many core developers), two entities who do not care about anything outside the GNOME _umbrella_.

Cinnamon created the Xapp library to avoid a similar doom as MATE. ElementaryOS has granite. Xfce has libxfceui and did something really wise and started migrating to non-deprecated widgets, which meant to embrace CSD and infuriated some trolls... I mean, users... but was what I think is a good choice, since most of Xfce developers are volunteers, and have to resources to create a fork of Gtk. Many trolls... I mean, users just think forking a project is as easy as clicking a button in Github, instead of realizing all the work required to actually make a succesful fork. So, the short-long-story: unlike MATE, Cinnamon and Xfce have a future, until Gtk5 kills X11 support.

So, hope this two paragraphs explain my personal point of view.

**Reply by tsujan** [reply]
Being based on Qt, LXQt will support all DE-agnostic X11 WMs as long as Qt supports X11.

As for `GLib`, it's a low-level C library, which LXQt uses for file and mimetype management. Unlike GTK, it has been quite reliable. Qt's file management is poor compared to it, and depending on KDE libraries wasn't an option from start. LXQt tries not to reinvent the wheel ;)

LXQt doesn't and won't depend on GTK or other GNOME libraries.

**Reply by darkshram** [reply]
What I meant with my comment was to avoid as much as possible to depend on third-party components.
